libpromises/cf3parse_logic.h

```diff
@@ -1058,4 +1058,50 @@ static inline void ParserHandleQuotedListItem()
     FREE_AND_NULL(P.currentstring);
 }
 
+static inline void ParserCheckPromiseLine()
+{
+    if (P.currentpromise == NULL)
+    {
+        return;
+    }
+
+    // The following promise types does not require any actions
+    static const char *const exceptions[] = {
+        "classes", "commands", "methods", "reports", "insert_lines",
+        "delete_lines", "build_xpath", "insert_tree" };
+
+    const char *const type = P.currenttype;
+    if (IsStringInArray(type, exceptions,
+                        sizeof(exceptions) / sizeof(exceptions[0])))
+    {
+        return;
+    }
```

Comment:
Should add logic for having an exception for custom promise types.

Reply:
https://github.com/cfengine/core/pull/5414#issuecomment-1918973158